pysatSeasons/plot.py

```diff
@@ -60,24 +70,30 @@ def scatterplot(inst, labelx, labely, data_label, datalim, xlim=None,
             ax1.set_ylim(ylim)
             ax2.set_ylim(ylim)
 
-    # norm method so that data may be scaled to colors appropriately
+    # Norm method so that data may be scaled to colors appropriately
     norm = mpl.colors.Normalize(vmin=datalim[0], vmax=datalim[1])
     p = [i for i in np.arange(len(figs))]
     q = [i for i in np.arange(len(figs))]
-    for i, inst in enumerate(inst):
-        for j, (fig, ax) in enumerate(zip(figs, axs)):
+    for linst in const.instruments:
+        for inst in linst:
             if not inst.empty:
-                check1 = len(inst.data[labelx]) > 0
-                check2 = len(inst.data[labely]) > 0
-                check3 = len(inst.data[data_label[j]]) > 0
-                if (check1 & check2 & check3):
-                    p[j] = ax[0].scatter(inst.data[labelx], inst.data[labely],
-                                         inst.data[data_label[j]], zdir='z',
-                                         c=inst.data[data_label[j]], norm=norm,
-                                         linewidth=0, edgecolors=None)
-                    q[j] = ax[1].scatter(inst.data[labelx], inst.data[labely],
-                                         c=inst.data[data_label[j]],
-                                         norm=norm, alpha=0.5, edgecolor=None)
+                for j, (fig, ax) in enumerate(zip(figs, axs)):
+                    check1 = len(inst.data[labelx]) > 0
+                    check2 = len(inst.data[labely]) > 0
+                    check3 = len(inst.data[data_label[j]]) > 0
+                    if check1 & check2 & check3:
```

Comment:
```suggestion
                    if check1 and check2 and check3:
```

Reply:
Though I recommend just putting the length evaluations in the if statement, since they aren't used elsewhere.